app/models/user.rb

```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  validates :name, presence: true
+  validates :bio, presence: true
+  validates :posts_counter, numericality: { only_integer: true, greated_than_or_equal_to: 0 }
+
   has_many :posts, foreign_key: 'author_id'
   has_many :likes, foreign_key: 'author_id'
   has_many :comments, foreign_key: 'author_id'
```

Comment:
- [ ] Kindly you still have the same error in your previous milestone 
![image](https://user-images.githubusercontent.com/65068771/161395769-81746f75-db20-4a8b-87c9-b819b5c57134.png)
> Kindly note that there is no column in your post table called author_id instead the user is referenced in the post table by the user_id column. Kindly fix this by either updating your post table(migration) to reference the user by the post_id or user the `user_id in your controllers and views.

## TO FIX THIS change these migrations

- [ ] chenage this :point_down: 
```rb
class AddUserRefToPosts < ActiveRecord::Migration[7.0]
  def change
    add_reference :posts, :user, null: false, foreign_key: true
  end
end
```
- to  this :point_down: 
```rb
class AddUserRefToPosts < ActiveRecord::Migration[7.0]
  def change
    add_reference :posts, :author, null: false, foreign_key: { to_table: :users }
  end
end
```


![image](https://user-images.githubusercontent.com/65068771/161396865-9bdc4429-e93a-4bd4-9c79-53cdd4290378.png)

- [ ] change this :point_down: 

```rb

class AddUserRefToComments < ActiveRecord::Migration[7.0]
  def change
    add_reference :comments, :user, null: false, foreign_key: true
  end
end

```
- to this :point_down: 
```.rb
class AddUserRefToComments < ActiveRecord::Migration[7.0]
  def change
    add_reference :comments, :author, null: false, foreign_key: { to_table: :users }
  end
end
``